### PR TITLE
MGMT-11814 Display FieldArray errors with getRichTextValidation

### PR DESCRIPTION
**Pull request by celdrake**
When using `validate={getRichTextValidation}` in a form, messages from FielArray were not displaying correctly. 
The problem is that `getRichTextValidation` generates an object where the paths are the individual fields, whereas `validate` is looking for the format of the parent field.

For example, for `diskEncryptionTangServers`, it needs the structure:

```
{diskEncryptionTangServers: [
{ url: 'Url error for Tang Server', thumbprint: 'Thumbprint error for Tang Server' }
]}
```
but it is seeing

`{ 'diskEncryptionTangServers[0].thumbprint': 'Some error}`

Now we are detecting the subfields and generating the needed format from the `diskEncryptionTangServers[0].thumbPrint` etc errors.

The PR also includes some clean-up of the Disk Encryption and other components.

### CODE REVIEW

**Comment by celdrake** [comment]
@rawagner @jtomasek I checked on CIM, and there could be one form which could be impacted. It's the `BMCForm`.
Could you check that it behaves correctly there?

Thanks!

**Review by ammont82 — DISMISSED** [review]

File: src/ocm/components/clusterConfiguration/OcmClusterDetailsFormFields.tsx
Comment (celdrake) [reply src/ocm/components/clusterConfiguration/OcmClusterDetailsFormFields.tsx:74]: My bad, that was for quicker testing but shouldn't be on the PR. Thanks for spotting it!

**Review by ammont82 — DISMISSED** [review]

**Review by jgyselov — DISMISSED** [review]
Approving but let's make sure we get a LGTM from CIM before we merge.

File: src/common/components/ui/formik/utils.ts
Comment (rawagner) [inline src/common/components/ui/formik/utils.ts:51]: types should start with capital letter

File: src/common/components/ui/formik/utils.ts
Comment (rawagner) [inline src/common/components/ui/formik/utils.ts:61]: we can remove the `// eslint-disable-line`
```suggestion
const fieldErrorReducer = (errors: innerError[]): fieldErrors => {
  return errors.reduce<fieldErrors>(
    (memo, { path, message }) => ({
      ...memo,
      [path]: (memo[path] || []).concat(message),
    }),
    {},
  );
};
```

**Review by rawagner — CHANGES_REQUESTED** [review]

File: src/common/components/ui/formik/utils.ts
Comment (celdrake) [reply src/common/components/ui/formik/utils.ts]: Fixed, thanks.
Will check if we can enforce it using `https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/naming-convention.md`

File: src/common/components/ui/formik/utils.ts
Comment (celdrake) [reply src/common/components/ui/formik/utils.ts]: Done

**Review by jgyselov — APPROVED** [review]

**Review by rawagner — APPROVED** [review]

**Pull request merged**
